### PR TITLE
Add rpy2 recipe

### PR DESCRIPTION
**Pull request by ceholden**
Adds `rpy2`, a Python library that allows you to run R code from within Python!

Getting this in would really helpful to me now that ICU creates conflicts with the `r` channel (:tada: for getting `r-base`!). Assuming this builds on all platforms, it should also address an issue recently reported in #1788.

One thought I had was about the minimum required specs. It's easy to just have `r-base` right now without more of R ported, but I'm open to suggestions if there's anything else that should be added eventually. Similarly, the Python requirements are sparse enough (no NumPy necessarily) to make `rpy2` not very useful! Again, I'm open to suggestions for additions but I'm assuming people will install NumPy/Pandas/etc as needed.

### CODE REVIEW

**Comment by johanneskoester** [comment]
The plan is to port the whole R channel to conda-forge. But this needs some time, and currently issues in the conda-forge build system prevent to quickly port things over.

**Comment by isuruf** [comment]
Since linux and osx packages are fine, I suggest skipping windows for now and getting the recipe merged.

File: recipes/rpy2/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    - gcc  # [not win]
+    - {{ native }}toolchain  # [win]
+    - readline  # [not win]
+    - python
+    - setuptools
+    - six
+    - r-base
+  run:
+    - {{ native }}gcc-libs  # [win]
+    - libgcc  # [win]
```
Comment (bgruening) [inline recipes/rpy2/meta.yaml]: [not win]

File: recipes/rpy2/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    - posix  # [win]
+    - gcc  # [not win]
+    - {{ native }}toolchain  # [win]
+    - readline  # [not win]
+    - python
+    - setuptools
+    - six
+    - r-base
+  run:
+    - {{ native }}gcc-libs  # [win]
```
Comment (bgruening) [inline recipes/rpy2/meta.yaml]: is this really needed?

File: recipes/rpy2/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    - rpy2.rinterface.tests
+    - rpy2.ipython
+    - rpy2.ipython.tests
+    - rpy2.rlike
+    - rpy2.rlike.tests
+    - rpy2.robjects
+    - rpy2.robjects.lib
+    - rpy2.robjects.lib.tests
+    - rpy2.robjects.tests
+  requires:
```
Comment (bgruening) [inline recipes/rpy2/meta.yaml]: Can you put the requires, above the imports?

File: recipes/rpy2/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+  license_file: gpl-2.0.txt
+  summary: 'Python interface to the R language (embedded R)'
+  description: |
+    rpy2 is an interface to R running embedded in a Python process.
+  doc_url: https://rpy2.readthedocs.io
+  dev_url: https://bitbucket.org/rpy2/rpy2
+
+extra:
+  recipe-maintainers:
+    - ceholden
```
Comment (bgruening) [inline recipes/rpy2/meta.yaml]: If you like please add `bgruening` and `johanneskoester` as well.

**Comment by ceholden** [comment]
Closing in favor #2950

**Pull request closed without merging**